include/triton/core/tritonserver.h

```diff
@@ -1883,6 +1883,30 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerDelete(
 TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerStop(
     TRITONSERVER_Server* server);
 
+/// Register a new model repository
+///
+/// \param server The inference server object.
+/// \param repository_path The full path to the model repository.
+/// \param name_mapping List of name_mapping parameters. Each path mapping has
+/// the model directory name as its key, overriden model name as its value.
+/// These map each directory to overwritten model names. This is for cases where
+/// the directory name does not match the model. \param model_count Number of
```

Comment:
Add comment about the special case where directory name is not provided (unknown) and user want the model inside has an overwritten name. This is the case we actually need to support.

Reply:
And newline for \param and \return